v2/client/invoker.go

```diff
@@ -65,24 +70,20 @@ func (r *receiveInvoker) Invoke(ctx context.Context, m binding.Message, respFn p
 			}
 		}
 
-		// protocol can manual ack by the result
+		// protocol can ack itself based on the result
 		if respFn == nil {
 			if !protocol.IsACK(result) {
-				err = m.Finish(result)
 				isFinished = true
 			}
-			return
+			return m.Finish(result)
 		}
 
-		var rm binding.Message
 		if resp != nil {
-			rm = (*binding.EventMessage)(resp)
+			respMsg = (*binding.EventMessage)(resp)
```

Comment:
while we are here, the cast should check that it was ok here with `respMsg, ok = ...` and then handle `!ok`

Reply:
`resp` is not an interface type so a type assertion won't work, and a cast returns only 1 value.